main.go

```diff
@@ -446,17 +511,43 @@ func MonitorLease(ctx context.Context, sm subnet.Manager, bn backend.Network, wg
 	}
 }
 
-func LookupExtIface(ifname string, ifregex string) (*backend.ExternalInterface, error) {
+func LookupExtIface(ifname string, ifregex string, ipStack int) (*backend.ExternalInterface, error) {
```

Comment:
It seems like for dual stack, we are using `ifregex` multiple times, so I would compile that expression.

```suggestion
func LookupExtIface(ifname string, ifregexS string, ipStack int) (*backend.ExternalInterface, error) {
	ifregex, err := regexp.Compile(ifregexS)
	if err != nil {
		return nil, fmt.Errorf("could not compile the IP address regex '%s': %w", ifregexS, err)
	}
```

Reply:
good idea